tests/test_group_graph.py

```diff
@@ -0,0 +1,163 @@
+        group_graph.compatibility(a, b)
+        exception_msg = excinfo.value.args[0]
+        assert exception_msg == "Tuples passed to compatibility() must have same size."
+
+
+def test_compatibility_objective_weights():
+    """ Test if objective_weights returns the objective weights of students """
+    a = [1, 1]
+    b = [0, 0.5]
+    objective_weights = (2.0, 1.0)
+    output = group_graph.compatibility(a, b, objective_weights)
+    expected_output = 1.75
+    assert output == expected_output
+
+
+def test_compatibility_objective_measures():
+    """ Test if objective_measures returns the objective measure of students """
+    a = [0, 1]
+    b = [0.75, 0.75]
+    objective_measures = ("avg", "match")
+    output = group_graph.compatibility(a, b, objective_measures=objective_measures)
+    expected_output = 0.375
+    assert output == expected_output
+
+
+def test_compatability_measure_average():
+    """ Test if measure of different student scores return an average """
+    a = [1, 1]
+    b = [0, 1]
+    output = group_graph.compatibility(a, b)
+    expected_output = 1.5
+    assert output == expected_output
+
+
+def test_compatability_measure_max():
+    """ Test if measure of different student scores return a maximum """
+    a = [1, 0]
+    b = [0, 0.5]
+    objective_measures = ("max", "max")
+    output = group_graph.compatibility(a, b, objective_measures=objective_measures)
+    expected_output = 1.5
+    assert output == expected_output
+
+
+def test_compatability_measure_min():
+    """ Test if measure of different student scores return a minimum """
+    a = [1, 0]
+    b = [0, 0.5]
+    objective_measures = ("min", "min")
+    output = group_graph.compatibility(a, b, objective_measures=objective_measures)
+    expected_output = 0
+    assert output == expected_output
+
+
+def test_compatability_measure_int():
+    """ Test if measure of different student scores are both equal """
+    a = [1, 0]
+    b = [0, 0.5]
+    objective_measures = ("match", "match")
+    output = group_graph.compatibility(a, b, objective_measures=objective_measures)
+    expected_output = 0
+    assert output == expected_output
+
+
+def test_compatability_measure_diff():
+    """ Test if measure of different student scores returns an absolute value difference """
+    a = [1, 0]
+    b = [0, 0.5]
+    objective_measures = ("diff", "diff")
+    output = group_graph.compatibility(a, b, objective_measures=objective_measures)
+    expected_output = 1.5
+    assert output == expected_output
+
+
+def test_group_graph_partition():
+    """
+    Test for using recursive Kernighan-Lin algorithm that checks the output of
+    the group_graph_partition function
+    """
+    students = [
+        ["one", 0, 0],
+        ["two", 0, 0.5],
+        ["three", 0.5, 0],
+        ["four", 0.75, 0.75],
+        ["five", 0.8, 0.1],
+        ["six", 0, 1],
+        ["seven", 1, 0],
+        ["eight", 1, 1],
+    ]
+    preference = {
```

Comment:
`group_graph_partition` expects this dictionary to have the form `student name: set of preferred student names` rather than `student name: preferred student name`. It doesn't make a difference in this test case, but it's worth keeping in mind

Reply:
I fixed the previous sets so that they now show the correct set of preferred student preferences.